### PR TITLE
Unwrap retries

### PR DESCRIPTION
**Pull request by sethhrbek**
## Overview
Adds code to retry unwraps in the case of failure.

It seems that the failure cases (currently) have to do with `estimateGas` failing, so this shouldn't cause issues with multiple TXs being reverted. I'm not 100% sure yet why the initial call to `estimateGas` fails as there isn't a good revert error message. Will do more sleuthing on this, but the retries solves the problem for the time being.

## NOTES
I moved some code into new files, so a lot of this code is not net-new.

## TODO
1. We could make the "max retries" an env var in the future if we want the arch service operator to control this var without having to edit the code.

2. Once the typechain defs are in the npm package, the service should be overhauled to use them as well as the types generated.

### CODE REVIEW

File: src/utils/blockchain/refresh-data.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  sarcoIds.map(async sarcoId => {
+    const sarcophagus = await web3Interface.viewStateFacet.getSarcophagus(sarcoId);
+    const archaeologist = await web3Interface.viewStateFacet.getSarcophagusArchaeologist(
+      sarcoId,
+      web3Interface.ethWallet.address
+    );
+
+    if (curseIsActive(sarcophagus, archaeologist)) {
+      const now = new Date().getTime() / 1000;
```
Comment (katien) [inline src/utils/blockchain/refresh-data.ts:34]: This is just a style thing but I find myself often getting confused by the switch between javascript's ms and solidity's seconds. On my branch I've been super explicit about units in my variable names, so I'd call this nowSeconds. It's not really a convention we've agreed on though so your call on whether you think it's helpful or redundant

File: src/utils/blockchain/refresh-data.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      const tooLateToUnwrap = now > endOfGracePeriod(sarcophagus, inMemoryStore.gracePeriod!);
+      if (tooLateToUnwrap) {
+        return;
+      }
+
+      const resurrectionTime = new Date(sarcophagus.resurrectionTime.toNumber() * 1000);
```
Comment (katien) [inline src/utils/blockchain/refresh-data.ts:41]: then I'd call this resurrectionTimeMs

File: src/utils/blockchain/unwrap.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    return await unwrapFn();
+  } catch (e) {
+    archLogger.warn(`Unwrap attempt ${depth + 1} failed, retrying....`);
+    if (depth > MAX_RETRIES) {
+      throw e;
+    }
+
+    await wait(INTERVAL_BETWEEN_RETRIES);
+
+    return unwrapSarcophagusWithRetry(unwrapFn, depth + 1);
```
Comment (katien) [inline src/utils/blockchain/unwrap.ts:23]: I like loops better for readability but that's also a style thing and up to you

File: src/utils/onchain-data.ts [hunk 4]
```diff
@@ -60,90 +57,13 @@ export async function getSarcophagiIds(web3Interface: Web3Interface): Promise<st
...
-        resurrectionTime,
-      });
-
-      // Schedule an uwrap job for each sarco this arch is cursed on. `scheduleUnwrap` will cancel existing
-      // schedules, so no duplicate jobs will be created.
-
-      // NOTE: It is marginally possible that the arch node and a sarco it's bonded to end up in a state where
-      // the sarco's resurrection time is past, but we're still within its grace period. In that case,
-      // scheduling a job won't work as resurrection time is in the past, so we'll want immediately attempt
-      // a rewrap in an effort to salvage the situation.
```
Comment (DarksightKellar) [inline src/utils/onchain-data.ts:110]: I feel like these comments are important to leave in. They've been removed in `refresh-data.ts:43`

**Review by DarksightKellar — APPROVED** [review]

**Review by katien — APPROVED** [review]

File: src/utils/blockchain/refresh-data.ts (hunk 1 above)
Comment (sethhrbek) [reply src/utils/blockchain/refresh-data.ts]: agree! this is updated to `nowSeconds`

File: src/utils/blockchain/refresh-data.ts (hunk 2 above)
Comment (sethhrbek) [reply src/utils/blockchain/refresh-data.ts]: thanks, good call -- updated!

File: src/utils/blockchain/unwrap.ts (hunk 3 above)
Comment (sethhrbek) [reply src/utils/blockchain/unwrap.ts:23]: I left as-is, just b/c i prefer loops when its best-case is the whole loop runs, whereas happy path here is it just runs once. but agree: readability is slightly degraded.

File: src/utils/onchain-data.ts (hunk 4 above)
Comment (sethhrbek) [reply src/utils/onchain-data.ts:110]: re-added!

**Pull request merged**